credsweeper/scanner/scan_type/pem_key_pattern.py

```diff
@@ -38,35 +40,39 @@ def run(cls, config: Config, rule: Rule, target: AnalysisTarget) -> Optional[Can
         """
         assert rule.pattern_type == rule.PEM_KEY_PATTERN, \
             "Rules provided to PemKeyPattern.run should have pattern_type equal to PEM_KEY_PATTERN"
-
-        if cls.is_pem_key(target.lines[target.line_num:], config):
+        global pem_pattern_check
+        if not pem_pattern_check:
+            pem_pattern_check = ValuePemPatternCheck(config)
```

Comment:
IMO `pem_pattern_check` can be defined as `cls.pem_pattern_check`.
Just define `cls.pem_pattern_check` variable as member.
``` python
class PemKeyPattern(ScanType):
        // ...
        cls.pem_pattern_check = ValuePemPatternCheck(config)
        // ...
       
        def run():
                // ...
```

Reply:
Thank you for the suggestion! Fix looks better.

Reply:
Thank you for a quick fix!
I've checked this change with cb13007.